### PR TITLE
Remove obsolete types from Debug/Trace/EventLog types

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

**Comment by BrennanConroy** [comment]
Fix your compile errors :D

File: src/Logging/Logging.Debug/src/DebugLogger.cs [hunk 1]
```diff
@@ -9,45 +9,31 @@ namespace Microsoft.Extensions.Logging.Debug
     /// <summary>
     /// A logger that writes messages in the debug output window only when a debugger is attached.
     /// </summary>
-    public partial class DebugLogger : ILogger
+    internal partial class DebugLogger : ILogger
```
Comment (BrennanConroy) [inline src/Logging/Logging.Debug/src/DebugLogger.cs]: You know what I'll say...

File: src/Logging/Logging.EventSource/src/EventSourceLogger.cs [hunk 2]
```diff
@@ -27,8 +27,8 @@ public EventSourceLogger(string categoryName, int factoryID, LoggingEventSource
         {
             CategoryName = categoryName;
 
-            // Default is to turn off logging
-            Level = LogLevel.None;
+            // Default is to turn on all the logging
+            Level = LogLevel.Trace;
```
Comment (BrennanConroy) [inline src/Logging/Logging.EventSource/src/EventSourceLogger.cs]: Oh?

File: src/Logging/Logging.EventSource/test/EventSourceLoggerTest.cs [hunk 3]
```diff
@@ -246,7 +219,7 @@ public void Logs_OnlyMessage_IfKeywordSet()
                     "E5MSG",
                     "InnerScopeStart",
                     "E6MSG",
-                    "InnerScopeStop",
+                    "InnerScopeStop",
```
Comment (BrennanConroy) [inline src/Logging/Logging.EventSource/test/EventSourceLoggerTest.cs]: nit

File: src/Logging/test/TraceSourceLoggerProviderTest.cs [hunk 4]
```diff
@@ -2,10 +2,10 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 #if NET461
+using Microsoft.Extensions.DependencyInjection;
```
Comment (BrennanConroy) [inline src/Logging/test/TraceSourceLoggerProviderTest.cs]: nit

File: src/Logging/Logging.Abstractions/src/NullScope.cs [hunk 5]
```diff
@@ -1,9 +1,9 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
 
-namespace Microsoft.Extensions.Logging.Abstractions.Internal
```
[the remark this answers is not in the document]
Comment (pakrym) [reply src/Logging/Logging.Abstractions/src/NullScope.cs]: Done

File: src/Logging/test/TraceSourceLoggerProviderTest.cs [hunk 6]
```diff
@@ -5,8 +5,9 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
-using Microsoft.Extensions.Logging.TraceSource;
 using Xunit;
+using Microsoft.Extensions.DependencyInjection;
```
Comment (BrennanConroy) [inline src/Logging/test/TraceSourceLoggerProviderTest.cs]: nit

File: src/Logging/test/TraceSourceLoggerProviderTest.cs [hunk 7]
```diff
@@ -5,8 +5,9 @@
...
+
```
Comment (BrennanConroy) [inline src/Logging/test/TraceSourceLoggerProviderTest.cs]: extra newline

**Comment by poke** [comment]
Is there a way we could make the `NullScope` public again? That’s actually one internal type that I have used in the past since I don’t want to implement a no-op disposable all the time in tests or logger providers that simply don’t have any scopes. Being able to share the same static instance was helpful there (even if it was internal).

**Comment by qub1n** [comment]
I have update nuget packages in my project and DebugLogger is now internal. What is the recommended replacement?

**Comment by BrennanConroy** [comment]
You should be adding debug as a provider in `ConfigureLogging` or on the logger factory:
https://docs.microsoft.com/aspnet/core/fundamentals/logging/?view=aspnetcore-3.1#add-providers

**Pull request merged**